apps/studio/tests/global-setup.ts

```diff
@@ -0,0 +1,155 @@
+import type { StartedTestContainer } from "testcontainers"
+import { stringify } from "superjson"
+import { GenericContainer, Wait } from "testcontainers"
+import { z } from "zod"
+
+// Specify necessary container configurations here
+const configurations: ContainerConfiguration[] = [
+  {
+    name: "database",
+    image: "postgres:15-alpine",
+    ports: [5432],
+    environment: {
+      POSTGRES_USER: "root",
+      POSTGRES_PASSWORD: "root",
+      POSTGRES_DB: "test",
+    },
+    wait: { type: "PORT" },
+  },
+]
+
+export const CONTAINER_INFORMATION_SCHEMA = z.array(
+  z.object({
+    name: z.string(),
+    host: z.string(),
+    ports: z.map(z.number(), z.number()),
+    configuration: z.object({
+      name: z.string(),
+      image: z.string(),
+      ports: z
+        .array(
+          z.union([
+            z.number(),
+            z.object({ container: z.number(), host: z.number() }),
+          ]),
+        )
+        .optional(),
+      environment: z.record(z.string(), z.string()).optional(),
+      wait: z
+        .union([
+          z.object({ type: z.literal("PORT"), timeout: z.number().optional() }),
+          z.object({
+            type: z.literal("LOG"),
+            message: z.string(),
+            times: z.number().optional(),
+            timeout: z.number().optional(),
+          }),
+          z.object({
+            type: z.literal("HEALTHCHECK"),
+            timeout: z.number().optional(),
+          }),
+        ])
+        .optional(),
+    }),
+  }),
+)
+
+type ContainerInformation = z.infer<typeof CONTAINER_INFORMATION_SCHEMA>[number]
+
+type ContainerConfiguration = ContainerInformation["configuration"]
+
+const setup = async (configurations: ContainerConfiguration[]) => {
+  const containerTemplates = configurations.map((configuration) => {
+    const { name, image, ports = [], environment, wait } = configuration
+    let container = new GenericContainer(image)
+
+    if (ports.length) {
+      container = container.withExposedPorts(...ports)
+    }
+
+    if (environment) {
+      container = container.withEnvironment(environment)
+    }
+
+    if (wait) {
+      const { type, timeout = 60 * 1000 } = wait
```

Comment:
super nit: maybe should be a `const` but htis is very low impact because it's in 1 place and also for tests